contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -143,6 +155,308 @@ contract OETHVaultCore is VaultCore {
+                "Queue pending liquidity"
+            );
+        }
+
+        // Store the updated claimed amount
+        withdrawalQueueMetadata.claimed = queue.claimed + request.amount;
+        // Store the request as claimed
+        withdrawalRequests[requestId].claimed = true;
+
+        emit WithdrawalClaimed(msg.sender, requestId, request.amount);
+
+        return request.amount;
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /// @notice Collects harvested rewards from the Dripper as WETH then
+    /// adds WETH to the withdrawal queue if there is a funding shortfall.
+    /// @dev is called from the Native Staking strategy when validator withdrawals are processed.
+    /// It also called before any WETH is allocated to a strategy.
+    function addWithdrawalQueueLiquidity() external {
+        // Stream any harvested rewards (WETH) that are available to the Vault
+        IDripper(dripper).collect();
+
+        _addWithdrawalQueueLiquidity();
+    }
+
+    /// @dev Adds WETH to the withdrawal queue if there is a funding shortfall.
+    /// This assumes 1 WETH equal 1 OETH.
+    function _addWithdrawalQueueLiquidity()
+        internal
+        returns (uint256 addedClaimable)
+    {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // Check if the claimable WETH is less than the queued amount
+        uint256 queueShortfall = queue.queued - queue.claimable;
+
+        // No need to get WETH balance if all the withdrawal requests are claimable
+        if (queueShortfall > 0) {
+            uint256 wethBalance = IERC20(weth).balanceOf(address(this));
+
+            // Of the claimable withdrawal requests, how much is unclaimed?
+            uint256 unclaimed = queue.claimable - queue.claimed;
+            if (wethBalance > unclaimed) {
+                uint256 unallocatedWeth = wethBalance - unclaimed;
+
+                // the new claimable amount is the smaller of the queue shortfall or unallocated weth
+                addedClaimable = queueShortfall < unallocatedWeth
+                    ? queueShortfall
+                    : unallocatedWeth;
+                uint256 newClaimable = queue.claimable + addedClaimable;
+
+                // Store the new claimable amount back to storage
+                withdrawalQueueMetadata.claimable = uint128(newClaimable);
+
+                // emit a WithdrawalClaimable event
+                emit WithdrawalClaimable(newClaimable, addedClaimable);
+            }
+        }
+    }
+
+    /***************************************
+                View Functions
+    ****************************************/
+
+    function _wethAvailable() internal view returns (uint256 wethAvailable) {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // Check if the claimable WETH is less than the queued amount
+        uint256 queueShortfall = queue.queued - queue.claimable;
```

Comment:
🟠 As clement mentioned, we use the equivalent of  `(queue.queued - queue.claimable) + (queue.claimable - queue.claimed)` twice. Instead of two variables, these can be combined into one set with `queue.queued - queue.claimed`

Reply:
this is an easy one. Not sure why I didn't pick it up when Clement mentioned it. Changed

Reply:
@naddison36 I think we have this function code twice in the code? Another one in OETHVaultAdmin. Need to update both of them?

Reply:
I've now updated both `_wethAvailable` functions on `OETHVaultCore` and `OETHVaultAdmin`.

Reply:
FYI, I think we should move the common internal functions between VaultAdmin and VaultCore into a common contract. I have implemented this PR https://github.com/OriginProtocol/origin-dollar/pull/2153